game.js

```diff
@@ -335,9 +335,11 @@ class Game {
   reset() {
     this.view = new View(this.squareSize);
     this.board = new Board(this.width, this.height);
-    this.isRunning = true;
-
+    this.isRunning = false;
+    this.generation = 0;
     this._addArmiesToBoard();
+    this.render();
+    this.eventListeners.dispatch(Events.NEW_GENERATION, this.generation);
```

Comment:
don't dispatch events here. also the generation is 0. The purpose of events is to notify when there is a change. There has been no change, you know the initial state is generation = 0

Reply:
This was done because when you reset, it doesn't dispatch an event to update the generation counter back to 0 (Also the reason for `this.render`. Namely, if the generation is at a non zero value, and I click reset, it will still display that non-zero value despite the fact that the game has been reset. I'll look for a different way to ensure it is consistent when reset occurs.